leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -7,61 +8,404 @@
+    */
     getFuncSegments: function(){
+
         evolvingFunc = this.get('evolvingFunction');
         if (evolvingFunc != null){
+            console.log(evolvingFunc.get('functionSegList'));
             return evolvingFunc.get('functionSegList');
         }
         return null;
-    }
+    },
+ 
+    /**
+     * Called whenever the user changes or removes the Initial Satisfaction Value
+     * in the Element Inspector. Called on Change for #init-sat-value 
+     * 
+     * Sets the refEvidence pair based on the chosen initValue
+     * 
+     * @param {*} initValue 
+     */
+    changeInitialSatValue: function(initValue) {
+
+        var intentionEval = graph.getUserEvaluationBBM(this.cid, '0');
+        intentionEval.set('assignedEvidencePair', initValue);
+        var funcSegList = this.getFuncSegments();
+        
+        if (this.get('evolvingFunction') != null) {
+            if (this.get('evolvingFunction').get('type') == 'C' || 
+                (this.get('evolvingFunction').get('type') == 'UD' && funcSegList[0].get('type') == 'C')) { 
+                    funcSegList[0].set('refEvidencePair', initValue); // Set first index of funcSegList to given initValue 
+            }
+            this.get('evolvingFunction').set('type', 'NT');
+            this.get('evolvingFunction').set('functionSegList', []);
+        }     
+    }, 
+ 
+    /**
+     * Sets the initial satisfaction value for this Intention to '(no value)'
+     */
+    removeInitialSatValue: function() {
+
+        this.changeInitialSatValue('(no value)');
+    },    
+ 
+    /**
+     * Returns the 4 digit representation for the
+     * Intention's initial satisfaction value
+     *
+     * @returns {String}
+     */
+    getInitialSatValue: function() {
+
+        var intentionEval = graph.getUserEvaluationBBM(this.cid, '0'); // Finding UserEvaluationBBM with same ID and startTP as this IntentionBBM 
+        if (typeof intentionEval == 'undefined'){
+            return '(no value)';
+        } else {
+            return intentionEval.get('assignedEvidencePair'); 
+        }
+    }, 
+ 
+    /**
+     * Resets evolving functions
+     */
+    removeFunction: function() {
+
+        this.removeAbsConstraint();
+        this.model.get('intention').set('evolvingFunction', null); // Set to null 
+    }, 
+ 
+    /**
+     * Clears all FuncSegments for the Intention's
+     * EvolvingFunction and adds new FuncSegments according to the current
+     * function type.
+     * Called when user changes the function type in Element Inspector 
+     * @param {String} funcType
+     */
+    setEvolvingFunction: function(funcType) {
+
+        this.set('evolvingFunction', new EvolvingFunctionBBM({type: funcType, functionSegList: []}));
+
+        // Since function changed, remove all current absolute constraints related to this intention
+        this.removeAbsConstraint();
+ 
+        // Add new absolute constraints if required
+        this.addAbsConstraint(funcType);
```

Comment:
I don't think I understand the use case for this function - why add an absolute constraint if you don't have a time point? @amgrubb ?